### ISSUE TITLE
keep branch version

### ISSUE THREAD

**Issue by pauljghosh** [cecom/pomutils#6]
pomutils has really saved us a lot of headache when dealing with multiple branches and maven versions.  We do have a request...  We try to keep versions per branch and sometimes git does the right thing and will update version during a merge since there is no conflict.  pomutils does not kick in since git did the merge without any issues.  We would like pomutils to have a feature where it keeps the version of the branch being merged into regardless of whether git creates a conflict or not... so if branch A is at version A.1 and branch B is at version B.1 and then branch A version is updated to A.2 and then merged into branch B the version should STAY at B.1 in branch B instead of being changed to A.2.  Can you do that for us?

**Comment by cecom** [comment]
What you describe is the following git history:

```
                       release/1.0.0
                      /
             o---o---o
            /
o---o---o---o          
             \        
              develop
```

If you would now merge release/1.0.0 into develop your develop branch would get the version of the release/1.0.0 branch. For git this is a fast forward and then the mergedriver isn't "called". Another reason for not calling the mergedriver is, if the file is only changed on one side of the merge. Git will then take the newer one. IMO there is no workaround for this scenarios to get the mergedriver triggered.

I also run into the same problem, but what i do is this: 

```
                 release/1.0.0
                /
               o
             /   \   git merge -s ours
o---o---o---o-----o            
                   \        
                    develop
```

When i create a release branch, the first commit on that branch is the adjustments of the `<version>`. I merge this commit back on the develop branch using `git merge -s ours`. Due to the fact, that the creation of a release branch is automated this is automaticly done.

In other words i really would like to implement what you are asking for, but this is a feature which has to be implemented in git :-)
